### PR TITLE
Update the auto_feedback_from_code_execution notebook to use code executor

### PR DESCRIPTION
**Pull request by ekzhu**
1. Update the auto_feedback_from_code_execution notebook to use code executor.
2. Removed the example that subclasses UserProxyAgent to implement IPython code executor. 
3. Add a custom code executor example to user guide for an IPython code executor.
4. Fix agentoptimizer notebook to make it displayed on the website. 
5. gitignore updates.

### CODE REVIEW

**Comment by skzhang1** [comment]
Thanks@ekzhu ! Now I know why I can not find AgentOptimizer in website.

File: notebook/agentchat_agentoptimizer.ipynb [hunk 1]
```diff
@@ -1,467 +1,466 @@
...
+        "\n",
+        "In this notebook, we introduce a new class, ‘AgentOptimizer’, which is able to improve the function list of one Assistant-UserProxy pair according to the historical conversation histories.\n",
+        "This feature would support agents in improving their ability to solve problems of the same type as previous tasks.\n",
+        "Specifically, given a set of training data, AgentOptimizer would iteratively prompt the LLM to optimize the existing function list of the AssistantAgent and UserProxyAgent with code implementation if necessary. It also includes two strategies, roll-back, and early-stop, to streamline the training process.\n",
+        "In the example scenario, we test the proposed AgentOptimizer in solving problems from the [MATH dataset](https://github.com/hendrycks/math). \n",
+        "\n",
+        "![AgentEval](../website/blog/2023-12-23-AgentOptimizer/img/agentoptimizer.png)\n",
```
Comment (sonichi) [inline notebook/agentchat_agentoptimizer.ipynb:19]: AgentOptimizer

File: notebook/agentchat_auto_feedback_from_code_execution.ipynb [hunk 2]
```diff
@@ -10,16 +10,13 @@
...
-    "Please find documentation about this feature [here](https://microsoft.github.io/autogen/docs/Use-Cases/agent_chat).\n",
-    "\n",
     "In this notebook, we demonstrate how to use `AssistantAgent` and `UserProxyAgent` to write code and execute the code. Here `AssistantAgent` is an LLM-based agent that can write Python code (in a Python coding block) for a user to execute for a given task. `UserProxyAgent` is an agent which serves as a proxy for the human user to execute the code written by `AssistantAgent`, or automatically execute the code. Depending on the setting of `human_input_mode` and `max_consecutive_auto_reply`, the `UserProxyAgent` either solicits feedback from the human user or returns auto-feedback based on the result of code execution (success or failure and corresponding outputs) to `AssistantAgent`. `AssistantAgent` will debug the code and suggest new code if the result contains error. The two agents keep communicating to each other until the task is done.\n",
     "\n",
     "````{=mdx}\n",
```
Comment (skzhang1) [inline notebook/agentchat_auto_feedback_from_code_execution.ipynb:15]: Does this instruction not render correctly?

Comment (ekzhu) [reply notebook/agentchat_auto_feedback_from_code_execution.ipynb:15]: The block will be rendered into a widget on the website.

**Review by skzhang1 — APPROVED** [review]
This PR looks good to me in my side.

**Pull request merged**